CHANGELOG.md

```diff
@@ -0,0 +1,342 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
+and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+### Added
+
+- Added contributor credits to README.
+
+- Added support for Ruby 2.3, 2.4 and 2.5.
+
+- Added support for Rails 5.0 and 5.1.
+
+- Support passing an array of metadata values.
+
+    ```ruby
+    DoubleEntry.transfer(
+      Money.new(20_00),
+      :from     => one_account,
+      :to       => another_account,
+      :code     => :a_business_code_for_this_type_of_transfer,
+      :metadata => { :key1 => ['value 1', 'value 2'], :key2 => 'value 3' },
+    )
+    ```
+
+- Allow partner account to be specified for aggregates.
+
+- Allow filtering aggregates by multiple metadata key/value pairs.
+
+### Changed
+
+- Replaced Machinist with Factory Girl in test suite.
+
+- Use `Money#positive?` and `Money#negative?`.
+
+- Back `DoubleEntry::Transfer::Set` and `DoubleEntry::Account::Set` with
```

Comment:
@orien I do not understand this part. Back ?

Reply:
I think this phrasing comes from _behind_:
what's behind the interface?
the implementation.

The `Transfer::Set` class is now backed by a `Hash`. The `Hash` is encapsulated by the `Transfer::Set`, you don't see it externally, however it powers much of its behaviour.

Reply:
Could you suggest an easier to understand alternative?